### PR TITLE
added JWT support

### PR DESCRIPTION
**Pull request by furthernull**

### CODE REVIEW

File: src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+public class UserLoginRequestDto {
+    @NotBlank
+    @Email
+    private String email;
+    @NotBlank
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java:13]:
```suggestion
    @NotBlank
```

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java:45]: Create constant for `"Bearer "`

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java:44]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```
Use constant from Spring

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

File: src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java (hunk 1 above)
Comment (Vasul-Pankiv) [reply src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java:13]: not fixed

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Pull request merged**